client/modules/components/src/Wikitext.svelte

```diff
@@ -131,6 +133,12 @@
     element = document.createElement("div")
     element.classList.add("wikitext-body", "wikitext")
     root.appendChild(element)
+
+    // if we're isolated, we need to add the ftml stylesheet to the shadow dom
+    const style = document.createElement("style")
+    style.className = "ftml-base-stylesheet"
```

Comment:
(nitpick) shouldn't this be an id not class?

Reply:
IDs are _supposed_ to be unique, so if there was multiple `Wikitext` components they would conflict. However, this is _technically inside a shadow DOM_, so I don't actually know if that matters. Regardless, this method works fine.

Reply:
Is it possible for there to be multiple of this component in one shadow DOM?

But regardless I guess this is fine as is now.

Reply:
lol in general I really try to avoid using ID unless it's super obviously a global-ish element, because using ID invites so many quirks.

for example, if you do:
```html
<body>
  <div id="foo">bar</div>
</body>
```

```js
// your editor will complain about this
console.log(foo.innerText) // but... this prints "bar"
```

yeah. blame IE

Reply:
W-what?

Alright fair enough....

Reply:
They added it to [the god damn spec](https://html.spec.whatwg.org/multipage/window-object.html#named-access-on-the-window-object) because the IE devs decided to make "named elements" automatically properties of the global object. For some reason. I guess it was convenient? Everyone else added the behavior because IE was the pack leader and big companies literally tested their sites only on IE. Then, when the spec was getting more standardized, because _everyone had the behavior_, it was kind of undeniably part of the de-facto spec already in place.

Welcome to HTML and JS.